From 5f3f2cccdd2d27367fe277114b6e3ad7fd42bfbc Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:42:59 +0530
Subject: [PATCH 0452/1851] v4l: xilinx-multi-scaler: Fix stop streaming

commit 0fb53c7043f0f48453f5d6ac0af437111f666258 from
https://github.com/Xilinx/linux-xlnx.git

Driver is setting bit on stop streaming which is incorrect
This patch corrects the functionality by clearing the bit
on stop streaming.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 0e9caa4da922..e899f5f1ee1a 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1438,9 +1438,9 @@ static void xm2msc_stop_streaming(struct vb2_queue *q)
 	xm2msc_return_all_buffers(chan_ctx, q, VB2_BUF_STATE_ERROR);
 
 	if (V4L2_TYPE_IS_OUTPUT(q->type))
-		xm2msc_set_chan_stream(chan_ctx, true, XM2MSC_CHAN_OUT);
+		xm2msc_set_chan_stream(chan_ctx, false, XM2MSC_CHAN_OUT);
 	else
-		xm2msc_set_chan_stream(chan_ctx, true, XM2MSC_CHAN_CAP);
+		xm2msc_set_chan_stream(chan_ctx, false, XM2MSC_CHAN_CAP);
 }
 
 static const struct vb2_ops xm2msc_qops = {
-- 
2.31.1

